module/VuFind/src/VuFind/Controller/HoldsTrait.php

```diff
@@ -87,8 +87,12 @@ public function holdAction()
         $validRequest = $catalog->checkRequestIsValid(
             $driver->getUniqueID(), $gatheredDetails, $patron
         );
-        if (!$validRequest) {
-            return $this->blockedholdAction();
+        if ((is_array($validRequest) && !$validRequest['valid']) || !$validRequest) {
```

Comment:
Why did you eliminate the call to blockedholdAction here, but not for storage retrieval or ILL requests?

If we don't call blockedholdAction, should we eliminate it entirely? Or is it better to add a parameter to the action so you can pass in the appropriate flash message instead of reproducing all of the functionality in-place here?